diff --git a/data/architectures/amd64.toml b/data/architectures/amd64.toml
--- a/data/architectures/amd64.toml
+++ b/data/architectures/amd64.toml
@@ -12,0 +13,4 @@ packages = [
+
+[packages.landscape-router]
+  scm = "local"
+  architecture = "amd64"
@@ -15,0 +20,8 @@
+
+[additional_repositories.nexttrace]
+  architecture = "amd64"
+  url = "https://github.com/nxtrace/nexttrace-debs/releases/latest/download"
+  distribution = "./"
+  components = ""
+  no_source = true
+  key = "mDMEaO7vHRYJKwYBBAHaRw8BAQdAXY+aVR+gfjDiaI1CQZXTiaOJI9ZOEziFVxQPheIyrDG0IW54dHJhY2Uub3JnIDxjb250YWN0QG54dHJhY2Uub3JnPoiTBBMWCgA7FiEEoqH7meOhHVsUbhcLNigaslezB/oFAmju7x0CGwMFCwkIBwICIgIGFQoJCAsCBBYCAwECHgcCF4AACgkQNigaslezB/rjdQD+JmXqnB3kmkKVQRfH7urC/zTkp2UkkLhUw+dykoCTggcBAK5qLKvUJN6Xu3TF0dEuBqVV9+14OkTbg4Ymaw7gUpEPuDgEaO7vHRIKKwYBBAGXVQEFAQEHQILPhsR2XsUYs7orwth28VQZRlXznHYsjCISbUb+UPgxAwEIB4h4BBgWCgAgFiEEoqH7meOhHVsUbhcLNigaslezB/oFAmju7x0CGwwACgkQNigaslezB/qHNwD/d3Yl/QhT/62Cghg575+XTEbnsrCIhazszdTd8NB4OFUA+wU421K4YjDrCAQy+C77YlRQ1pxFv3cYIMHn3H9qTEUM"
